lsp/src/main/kotlin/org/gradle/declarative/lsp/DeclarativeTextDocumentService.kt

```diff
@@ -403,9 +397,9 @@ private fun computeTypedPlaceholder(
     type: DataTypeRef,
     analysisSchema: AnalysisSchema
 ): String {
-    val resolvedType = SchemaTypeRefContext(analysisSchema).resolveRef(type)
-    return when (resolvedType) {
+    return when (val resolvedType = SchemaTypeRefContext(analysisSchema).resolveRef(type)) {
         is DataType.BooleanDataType -> "\${$index|true,false|}"
+        is EnumClass -> "\${$index|${resolvedType.entryNames.joinToString(",")}|}"
```

Comment:
⭐ I am amazed that this is only what we need to change to have full enum support essentially. Nice job!